internal/cltest/mocks.go

```diff
@@ -713,3 +715,102 @@ type MockPasswordPrompter struct {
 func (m MockPasswordPrompter) Prompt() string {
 	return m.Password
 }
+
+type EventWebsocketServer struct {
+	*httptest.Server
+	mutex       *sync.RWMutex // shared mutex for safe access to arrays/maps.
+	t           *testing.T
+	connections []*websocket.Conn
+	Connected   chan struct{}
+	Received    chan string
+	URL         *url.URL
+}
+
+func NewEventWebsocketServer(t *testing.T) (*EventWebsocketServer, func()) {
+	server := &EventWebsocketServer{
+		mutex:     &sync.RWMutex{},
+		t:         t,
+		Connected: make(chan struct{}, 1), // have buffer of one for easier assertions after the event
+		Received:  make(chan string, 100),
+	}
+
+	server.Server = httptest.NewServer(http.HandlerFunc(server.handler))
+	u, err := url.Parse(server.Server.URL)
+	if err != nil {
+		t.Fatal(err)
+	}
+	u.Scheme = "ws"
+	server.URL = u
+	return server, func() {
+		server.Close()
+	}
+}
+
+var upgrader = websocket.Upgrader{
+	CheckOrigin: func(r *http.Request) bool { return true },
+}
+
+func (wss *EventWebsocketServer) handler(w http.ResponseWriter, r *http.Request) {
+	var err error
+	conn, err := upgrader.Upgrade(w, r, nil)
+	if err != nil {
+		wss.t.Fatal(err)
+	}
+
+	wss.addConnection(conn)
```

Comment:
This is inside an httptest server, provided by [golang httptest](https://golang.org/pkg/net/http/httptest/), so my intuition led me to assume a webserver will have multiple connections, and implement as such. You're right though that our tests only have one connection at the moment. 

We need the `conn` to `wss.WriteCloseMessage` to emulate a server restart, so only a bool wouldn't do, we would need at the very least a `w.conn`. And with that shared `w.conn`, I wouldn't feel safe merely overwriting it on every subsequent connection. 

I don't think there's any advantage removing `wss.connections` considering it's already done. Let me know if you have any suggestions to improve this.

Reply:
It feels like YAGNI / complications that might slow down future work.

Reply:
This one I'm a little more attached to because I'm uncomfortable making an N web handler collapse into a shared 1 `wss.conn` that would get clobbered.